### ISSUE TITLE
Please add a way to 

### ISSUE THREAD

**Issue by dchawisher** [janeczku/calibre-web#2370]
**Is your feature request related to a problem? Please describe.**
I would like to transition from desktop calibre to calibre-web. My calibre workflow involves a set of metadata processing rules that result in all my books being members of a handful of Categories (e.g. the book *Gridlinked* might be `fiction`, `science-fiction`, `thriller`). I then use the Kobo extended driver to send those Categories to my Kobo as Collections. The result is a well-organized Kobo library that is easy to browse even with hundreds of books on the device.

With calibre-web, I cannot sync Categories (or any Calibre metadata) to Kobo as Collections. Instead I must manually add each ebook to the appropriate Shelves. Because my ebook library is substantial, and because managing Shelves manually is cumbersome, this is infeasible. 

**Describe the solution you'd like**
Any of the following:

- The ability to manage Shelves in calibre. (I understand that this is very unlikely.) 
- The ability to assign shelves automatically in calibre-web based on calibre metadata. (For example: each of the comma-separated values in the 'tags' field becomes a Shelf)
- The ability to use calibre metadata, rather than Shelves, as the "Collection" for purposes of Kobo syncing.

**Additional context**
The cumbersome management of Shelves is a common concern.
Other issues requesting a way to automatically set Shelves from Calibre metadata (from categories, custom columns, or other solutions): #2316, #1980, #1649, #1185, #809, #760, #136
Other issues related to streamlining Shelves management: #2272, #1914, #1888, #1799

**Comment by jeff47** [comment]
Agreed - I would love to be able to have things sync to my Kobo by setting a tag in Calibre, rather than having to first import the book to Calibre and then open Calibre-Web to add it to the synced shelf.

**Comment by adamshand** [comment]
Yes please!  It would be wonderful to be able to control what gets sync'd to my Kobo from Calibre Desktop.

**Comment by supfors** [comment]
When playing with python pandas the other day I figured I might as well write a script to create/update calibre-web shelves based on my calibre tags. Hope it helps a little.

[Source](https://github.com/supfors/c2cw_shelf_sync)

**Comment by new-usemame** [comment]
This works in [Calibre-Web-NextGen](https://github.com/new-usemame/Calibre-Web-NextGen), a community-maintained CWA build — currently [v4.1.30](https://github.com/new-usemame/Calibre-Web-NextGen/releases/tag/v4.1.30). Build a Magic Shelf from a rule over tags, series, authors, ratings or read state, mark it for Kobo sync, and the device gets a shelf that maintains itself from your calibre metadata instead of one you curate by hand.

```
docker pull ghcr.io/new-usemame/calibre-web-nextgen:latest
```

Heads-up if you run plain calibre-web rather than CWA: this build also brings auto-ingest, kepubify and KOReader sync. Anything not working, the fork tracker is https://github.com/new-usemame/Calibre-Web-NextGen/issues.
